src/ingest_validation_tools/submission.py

```diff
@@ -106,7 +107,7 @@ def _get_single_tsv_external_errors(self, type, path):
         errors = {}
         rows = _get_tsv_rows(path)
         if not rows:
-            errors['Warning'] = f'File has no data rows.'
+            errors['Warning'] = f'File {Path(path).name} has no data rows.'
```

Comment:
```suggestion
            errors['Warning'] = f'File has no data rows.'
```
Contextual information like filename is supplied at a higher level: it doesn't need to be repeated here. [For example](https://github.com/hubmapconsortium/ingest-validation-tools/blob/ef78c325b0e07b895f336a91c53d4089a7e55c87/dataset-examples/bad-no-such-type/README.md):
```
Metadata TSV Errors:
  dataset-examples/bad-no-such-type/submission/fake-metadata.tsv (as fake):
    Internal:
      No such file or directory: fake.yaml
    External:
      Warning: File has no data rows.
```

If you're concerned about the human-readable-flat list, that was handled in [PR #389](https://github.com/hubmapconsortium/ingest-validation-tools/pull/389#issuecomment-715609783).

General theme is that I'd prefer not to duplicate information across the response data structure.

Reply:
Fair enough, but the f'' string generates a PEP8 error from flake8 if there is no {} expression within the quoted string.  If {Path(path).name} is removed, the leading 'f' should be also.